### PR TITLE
[kube-prometheus-stack] Update port for metric of etcd

### PR DESCRIPTION
**Pull request by nobiit**
#### What this PR does / why we need it:
Fix port for metric etcd

#### Special notes for your reviewer:
https://github.com/kubernetes/kubernetes/blob/dafa55be106730eaacb740e20b999b84848c6ac5/cmd/kubeadm/app/constants/constants.go#L95-L96

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -1284,8 +1284,8 @@ kubeEtcd:
   ##
   service:
     enabled: true
-    port: 2379
-    targetPort: 2379
+    port: 2381
+    targetPort: 2381
```
Comment (monotek) [inline charts/kube-prometheus-stack/values.yaml:1288]: Maybe we should have some logic here using the right port depnedeing on the k8s version?
The values of the ports could be empty and only used when actualy set.

Comment (nobiit) [reply charts/kube-prometheus-stack/values.yaml:1288]: A pretty good idea, maybe I'll give it a try :D

Comment (nobiit) [reply charts/kube-prometheus-stack/values.yaml:1288]: Hey @monotek , I checked kubernetes git documentation and history (via blame), it seems that etcd metrics was always running on port 2381.
https://github.com/kubernetes/kubernetes/blob/96a32be18dfbd38ff5cb1df4d68528e2cf226958/cmd/kubeadm/app/constants/constants.go#L96 

I checked kubernetes git documentation and history (via blame), it seems that etcd metrics was always running on port 2381

Comment (monotek) [reply charts/kube-prometheus-stack/values.yaml:1288]: Not sure about that: https://etcd.io/docs/v3.1/op-guide/monitoring/

I don't have time to check further. Have only my mobile phone the next 4 weeks...

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.4.1
+version: 39.4.2
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 40.0.1
```

File: charts/kube-prometheus-stack/values.yaml (hunk 1 above)
Comment (monotek) [reply charts/kube-prometheus-stack/values.yaml:1288]: just tested it on a kind cluster. it is localhost:2381/metrics

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Comment by nobiit** [comment]
Thanks @monotek

**Pull request merged**